central/cloudsources/service/service_impl.go

```diff
@@ -0,0 +1,271 @@
+package service
+
+import (
+	"context"
+
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/cloudsources/datastore"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/auth/permissions"
+	"github.com/stackrox/rox/pkg/endpoints"
+	"github.com/stackrox/rox/pkg/errorhelpers"
+	"github.com/stackrox/rox/pkg/errox"
+	"github.com/stackrox/rox/pkg/grpc/authz"
+	"github.com/stackrox/rox/pkg/grpc/authz/perrpc"
+	"github.com/stackrox/rox/pkg/grpc/authz/user"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/search/paginated"
+	"github.com/stackrox/rox/pkg/secrets"
+	"github.com/stackrox/rox/pkg/sliceutils"
+	"github.com/stackrox/rox/pkg/uuid"
+	"google.golang.org/grpc"
+)
+
+const (
+	maxPaginationLimit = 1000
+)
+
+var (
+	_ v1.CloudSourcesServiceServer = (*serviceImpl)(nil)
+
+	authorizer = perrpc.FromMap(map[authz.Authorizer][]string{
+		user.With(permissions.View(resources.Integration)): {
+			"/v1.CloudSourcesService/CountCloudSources",
+			"/v1.CloudSourcesService/GetCloudSource",
+			"/v1.CloudSourcesService/ListCloudSources",
+			"/v1.CloudSourcesService/TestCloudSource",
+		},
+		user.With(permissions.Modify(resources.Integration)): {
+			"/v1.CloudSourcesService/PostCloudSource",
+			"/v1.CloudSourcesService/PutCloudSource",
```

Comment:
Do you have a suggestion that distinguishes between `Put` and `Patch`? I don't think `UpdateCloudSource` makes that clear (although in this case we don't have a `Patch`, but in case we would want to add it later - how it be named?).

Reply:
Put applies the update to the object and expects the whole object in the request. So, I think the confusion around `UpdateCloudSource` is uncalled for. 

In order to use a generic method name, `patch,` the method should express the field that needs to be updated through field masking. Therefore, until we have that capability. The patch method names should be explicit such as `DisableCloudSource` and such.